### DISCUSSION TITLE
tree-sitter grammar change/build reload?

### DISCUSSION THREAD

**Discussion by EmranMR** [helix-editor/helix#10115] [Q&A] [answered]
Hi,

I am trying to develop a `tree-sitter` grammar in `helix` however the only way I can reflect the changes after `hx --grammar build` is by quitting `helix` and restarting it... Basically the `:config-reload` has no effect for parsing the target language, ie `:tree-sitter-subtree` will show the old information...

I am just double checking that there are no better alternatives? are there any better workflow? It is quite painful if not because:
- After each restart the buffer will be lost
- can not undo/redo
- The workflow will be pretty slow, due to none stop restarting (especially if trialing changes quickly)
Thanks!

**Comment by the-mikedavis** [comment, marked as answer]
Parsers and queries are lazily loaded and only loaded once. There isn't an easy way to change this even if you want to modify the source locally. I would recommend relying more on tree-sitter's parsing and highlighting unit tests and using the playground `tree-sitter playground` for grammar development instead

**Reply by EmranMR** [reply]
Ah thanks good to know! Yea unfortunately I am trying to get the `indents.scm` working, which requires some fine tuning of the grammar as well. Otherwise I usually stick to the `tree-sitter-cli`!
